Foundation/NSFileManager.swift

```diff
@@ -417,7 +417,38 @@ public class NSFileManager : NSObject {
     }
     
     public func copyItemAtPath(srcPath: String, toPath dstPath: String) throws {
-        NSUnimplemented()
+        let fd_from = open(srcPath, O_RDONLY)
+        if fd_from < 0 {
+            throw _NSErrorWithErrno(errno, reading: true, path: dstPath, url: nil, extraUserInfo: nil)
+        }
+        defer { precondition(close(fd_from) >= 0) }
+        let permission = (try! attributesOfItemAtPath(srcPath)[NSFilePosixPermissions] as! NSNumber).unsignedShortValue
+        let fd_to = open(dstPath, O_WRONLY | O_CREAT | O_EXCL, permission)
+        if fd_to < 0 {
+            throw _NSErrorWithErrno(errno, reading: false, path: dstPath, url: nil, extraUserInfo: nil)
+        }
+        defer { precondition(close(fd_to) >= 0) }
+        
+        var buf = [UInt8](count: 4096, repeatedValue: 0)
+        
+        while true {
+            let nread = read(fd_from, &buf, buf.count)
```

Comment:
Using `sendfile` on Linux results in error

```
error: use of unresolved identifier 'sendfile'
```

I gather that this doesn't come free with Glibc?